### ISSUE TITLE
Feedback API questions

### ISSUE THREAD

**Issue by lxylxy123456** [lxylxy123456/ngshare#11]
What happens when an instructor upload feedbacks for a same submission? Should the files be overwritten?

**Comment by Lawrence37** [comment]
Yes.

**Comment by lxylxy123456** [comment]
When posting feedback, are we always posting on the latest submission? (I changed the API to allow posting on any submission)

When downloading feedback, are we always downloading the latest submission's feedback? What if instructor uploaded feedback then student submitted again?

**Comment by aalmanza1998** [comment]
feedback is released for the latest submission.

**Comment by lxylxy123456** [comment]
Then the only remaining question is: What if instructor uploaded feedback then student submitted again?

**Comment by aalmanza1998** [comment]
if the instructor released feedback and the student submitted again, the instructor would need to see it, auto grade it and generate feedback again. The student would then also have to fetch the feedback again. The student can see feedback for different submissions 
<img width="1220" alt="Screen Shot 2020-02-22 at 9 50 42 PM" src="https://user-images.githubusercontent.com/52981614/75104389-63d05900-55bd-11ea-9665-4fdd30471aa3.png">

**Comment by lxylxy123456** [comment]
So it is basically being able to download feedback based on time stamp? I will go change that part of API and vserver.

**Comment by aalmanza1998** [comment]
When feedback is released it is named using a `notebook_hash` which is generated using the name of the submission and a unique key (formed by course id, assignment id, notebook id, student id, and timestamp). Here's an example of the name of a feedback file: 
`232a28542e202cae455ace6f99a5c54b.html`

it fetches feedback by generating that same hash value and looking for it:
`feedbackpath = os.path.join(self.outbound_path, '{0}.html'.format(nb_hash))`


so in short, yes it needs the time stamp.

**Comment by lxylxy123456** [comment]
Do we need to keep all feedbacks generated on the same submission? (If so, it will be a huge project)

**Comment by lxylxy123456** [comment]
I have updated API and vserver. See https://github.com/lxylxy123456/ngshare/commit/a87d329a07a82fcd785713fba22a293d837e31c2

**Comment by aalmanza1998** [comment]
No, if the instructor releases new feedback for the same submission it replaces the old one and shows the student that there is new feedback to fetch for that submission. 
<img width="571" alt="Screen Shot 2020-02-22 at 10 13 59 PM" src="https://user-images.githubusercontent.com/52981614/75104741-819fbd00-55c1-11ea-9983-e723933e4fd3.png">
